File: fortran/envisat/absorbit.f
      program absorbit

      implicit none

      include "coda.inc"

      character*1024 filename
      character*32 product_class
C use 'integer*8 pf' for 64-bit
      integer pf
C use 'integer*8 cursor' for 64-bit
      integer cursor
      integer*4 abs_orbit
      integer result

      write(*,*) 'Name of the product file:'
      read(*,'(A1024)') filename

      result = coda_init()
      if (result .ne. 0) then
        call handle_coda_error()
      end if

      result = coda_open(filename, pf)
      if (result .ne. 0) then
        call handle_coda_error()
      end if

      result = coda_get_product_class(pf, product_class)
      if (product_class(1:7) .ne. 'ENVISAT') then
        write(*,*) 'Error: file is not an ENVISAT file'
        stop
      end if

      cursor = coda_cursor_new()

      result = coda_cursor_set_product(cursor, pf)
      if (result .ne. 0) then
        call handle_coda_error()
      end if

      result = coda_cursor_goto_record_field_by_name(cursor, 'mph')
      if (result .ne. 0) then
        call handle_coda_error()
      end if

      result = coda_cursor_goto_record_field_by_name(cursor,
     &    'abs_orbit')
      if (result .ne. 0) then
        call handle_coda_error()
      end if

      result = coda_cursor_read_int32(cursor, abs_orbit)
      if (result .ne. 0) then
        call handle_coda_error()
      end if

      write(*,*) 'absolute orbit: ', abs_orbit

      call coda_cursor_delete(cursor)

      result = coda_close(pf)

      call coda_done()

      end program


      subroutine handle_coda_error

      implicit none

      include "coda.inc"

      integer err
      character*75 errstr

      err = coda_get_errno()
      call coda_errno_to_string(err, errstr)
      write(*,*) 'Error: ' // errstr
      stop

      end subroutine

File: c/envisat/absorbit.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "coda.h"

int main(int argc, char *argv[])
{
    const char* product_class;
    coda_product *pf;
    coda_cursor cursor;
    int32_t abs_orbit;
    int result;

    if (argc != 2)
    {
        fprintf(stderr, "Usage: %s <envisat file>\n", argv[0]);
        exit(1);
    }

    coda_init();

    result = coda_open(argv[1], &pf);
    if (result != 0)
    {
        fprintf(stderr, "Error: %s\n", coda_errno_to_string(coda_errno));
        exit(1);
    }

    result = coda_get_product_class(pf, &product_class);
    if (result != 0)
    {
        fprintf(stderr, "Error: %s\n", coda_errno_to_string(coda_errno));
        exit(1);
    }
    if (strncmp(product_class, "ENVISAT", 7) != 0)
    {
        fprintf(stderr, "Error: file %s is not an ENVISAT product\n", argv[1]);
        exit(1);
    }

    result = coda_cursor_set_product(&cursor, pf);
    if (result != 0)
    {
        fprintf(stderr, "Error: %s\n", coda_errno_to_string(coda_errno));
        exit(1);
    }

    result = coda_cursor_goto_record_field_by_name(&cursor, "mph");
    if (result != 0)
    {
        fprintf(stderr, "Error: %s\n", coda_errno_to_string(coda_errno));
        exit(1);
    }

    result = coda_cursor_goto_record_field_by_name(&cursor, "abs_orbit");
    if (result != 0)
    {
        fprintf(stderr, "Error: %s\n", coda_errno_to_string(coda_errno));
        exit(1);
    }

    result = coda_cursor_read_int32(&cursor, &abs_orbit);
    if (result != 0)
    {
        fprintf(stderr, "Error: %s\n", coda_errno_to_string(coda_errno));
        exit(1);
    }

    printf("absolute orbit: %d\n", (int)abs_orbit);

    coda_close(pf);

    coda_done();

    return 0;
}
